features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/TreeViewTraversal.kt

```diff
@@ -38,11 +39,16 @@ internal class TreeViewTraversal(
         val resolvedWireframes: List<MobileSegment.Wireframe>
 
         // try to resolve from the exhaustive type mappers
-        val exhaustiveTypeMapper = mappers.findFirstForType(view::class.java)
+        val mapper = mappers.findFirstForType(view::class.java)
 
-        if (exhaustiveTypeMapper != null) {
-            val queueableViewMapper = QueueableViewMapper(exhaustiveTypeMapper, recordedDataQueueRefs)
-            traversalStrategy = TraversalStrategy.STOP_AND_RETURN_NODE
+        if (mapper != null) {
+            val queueableViewMapper =
+                QueueableViewMapper(mapper, recordedDataQueueRefs)
+            traversalStrategy = if (mapper is NonExhaustiveTypeMapper) {
+                TraversalStrategy.TRAVERSE_ALL_CHILDREN
```

Comment:
> [!NOTE]
> I'm confused about the naming of the interface: It says the mapper is Non exhaustive, but it would imply using a `TRAVERSE_ALL_CHILDREN` strategy. Can we be a bit more explicit about what is being non exhaustive here?

Reply:
Until now we've referred to mappers that return a node whenever they map to a specific class as being "exhaustive", and so the naming follows from this. How about "TraverseAllChildrenMapper"?